### DISCUSSION TITLE
'series' collection

### DISCUSSION THREAD

**Discussion by gasatrya** [11ty/eleventy#2523] [Q&A]
I'd like to create something like this:

```
---
tags: eleventy
series: eleventy
---
```

Then display all articles under eleventy `series` under each article that has the same series. Is there a way to do that?

**Comment by btrem** [comment]
Maybe something like this would do the job?:

In .eleventy.js, add the following block:

    eleventyConfig.addCollection('postsBySeries', function(collectionApi) {

        let series;
        let postsBySeries = {};
        let posts = collections.all;

        posts.forEach(function(post) {

            if (post.data.series) {

                series = post.data.series;

                // create a new array key with the series value if it doesn't already exist
                if (!postsBySeries[series]) {
                    postsBySeries[series] = [];
                }
                // add post to the year array key
                postsBySeries[series].push(post);

            }


        });

        return postsBySeries;
    });


Then, in a post, if series is defined in frontmatter, show the contents of that key from postsBySeries.
